### ISSUE TITLE
Running nomad/terraform/aws/ packer with an ubuntu 18.04 ami

### ISSUE THREAD

**Issue by rlewkowicz** [hashicorp/nomad#6137]
So I'm opening this as a bug, because the current packer for the demo terraform stuff doesn't work with 18.04 and I think 16.04 is EOL. I had a hell of a time trying to get resolved to play nice. You could define your bind IPs, but I want 0.0.0.0 and that wont bind with resolved on 127.

Solution:
Grab your ubuntu 18.04 ami
```
aws ec2 describe-images  --filters Name="name,Values=ubuntu/images/hvm-ssd/ubuntu*18.04*" --owners 099720109477 --query 'sort_by(Images, &CreationDate)[].[Name, ImageId,OwnerId]' --output text    | tail -1
```

Then in client.sh and server.sh above the systemctl block you add stub listener stuff :
```
sudo bash -c "echo 'DNSStubListener=no' >> /etc/systemd/resolved.conf"
sudo systemctl restart systemd-resolved
sudo systemctl enable consul.service
sudo systemctl start consul.service
```

I'd make a pull, but who knows if that'll get accepted. Too, feel free to close this if we don't want to fix it

**Comment by tgross** [comment]
Hi @rlewkowicz, thanks for reporting this issue!

As you've noted, the demo stack is all on Ubuntu 16.04. It looks like `systemd-resolved` wasn't yet the resolver for 16.04, and you can see in [`terraform/shared/scripts/client.sh#L60-L63`](https://github.com/hashicorp/nomad/blob/ac44517f504638a7f4b4938eb2ec333af9846717/terraform/shared/scripts/client.sh#L60-L63) where we've had to make some changes to get DNS working for Docker containers in our provisioning scripts.

Because Ubuntu 16.04 doesn't go EOL until April 2021 we might not immediately jump to getting the demo environment updated. But getting provisioning scripts working for both 16.04 and 18.04 would be super valuable, and if you were willing to make a patch I'm sure we'd be happy to accept it!

**Comment by rlewkowicz** [comment]
Unfortunately I moved on from nomad and am just swamped at the moment
